src/test/java/net/datafaker/providers/base/BaseFakerTest.java

```diff
@@ -72,22 +74,29 @@ protected void testProviderList(TestSpec testSpec, TestInfo testInfo) {
 
     protected Collection<TestSpec> providerListTest() {
         // dummy test since parameterized test requires non-empty collection
-        return Collections.singleton(new TestSpec(null, null));
+        return Collections.singleton(new TestSpec(null, null, null));
     }
 
     protected static class TestSpec {
         private final Supplier<?> supplier;
         private final String key;
         private final boolean isDummy;
+        @SuppressWarnings("unused")
+        private final String regex;
 
-        private TestSpec(Supplier<?> supplier, String key) {
+        private TestSpec(Supplier<?> supplier, String key, String regex) {
             this.supplier = supplier;
             this.key = key;
             this.isDummy = key == null || supplier == null;
+            this.regex = regex;
         }
 
         public static TestSpec of(Supplier<?> supplier, String key) {
-            return new TestSpec(supplier, key);
+            return new TestSpec(supplier, key, "");
+        }
+
+        public static TestSpec of(Supplier<?> supplier, String key, String regex) {
+            return new TestSpec(supplier, key, "");
```

Comment:
shouldn't it be like that?
```suggestion
            return new TestSpec(supplier, key, regex);
```
or even more i would suggest `Pattern.compile` to compile it during test input init

Reply:
Yup I think that was a copy/paste mistake 🤦‍♂️